src/linear_algebra/finite_dimensional.lean

```diff
@@ -312,6 +312,17 @@ iff.trans (by { rw ← findim_eq_dim, norm_cast }) (@dim_pos_iff_nontrivial K V
 lemma findim_pos [finite_dimensional K V] [h : nontrivial V] : 0 < findim K V :=
 findim_pos_iff.mpr h
 
+/-- `has_one` instance for `fin k` where `k` is the dimension of a nontrivial finite-dimensional
+vector space.  This is primarily useful so that `nonempty` can be deduced, by typeclass inference.
+-/
+noncomputable instance has_one_findim [nontrivial V] [finite_dimensional K V] :
+  has_one (fin (findim K V)) :=
```

Comment:
Wouldn't you rather use `has_zero`? When the dimension is `1`, then we are considering `fin 1 = {0}`. Of course it has a one (which just happens to be zero), but it has a better zero.

Reply:
You're right (I had just I blindly followed the output of `show_term` for my nonempty instance).  Fixed!